openedx/openedx-platform
# [BD-03] [TNL-2624] [BB-3168] Piazza Plugin for discussions

Adds a plugin to enable Piazza discussions in the discussions tab. 
Build on https://github.com/edx/edx-platform/pull/24584


**Dependencies**: https://github.com/edx/edx-platform/pull/24584

**Screenshots**: Always include screenshots if there is any change to the UI.

**Sandbox URL**: sandbox is being provisioned.
- LMS: https://pr25552.sandbox.opencraft.hosting/
- Studio: https://studio.pr25552.sandbox.opencraft.hosting/


**Test Instructions**:
- After checking out this branch run `pip install -e .` in the edx-platform directory in the lms and studio containers so that changes to setup.py are picked up. 
- Visit http://localhost:18000/admin/discussions/discussionproviderconfig/ and add a new provider config for Piazza. 
  - Give it any name
  - Select "Piazza" as the discussion provider
  - Set the config to ```{"consumer_key":"piazza.sandbox","consumer_secret":"test_only_secret"}```
  - Leave the rest empty and save. 
- Visit http://localhost:18000/admin/discussions/learningcontextdiscussionconfig/ and add a new object
  - Enter a valid course id in the Learning Context field
  - Enabled should be checked
  - Select the new provider. 
  - Leave the overrides blank.
- Visit the course page on the LMS
-  Make sure the Discussion tab is visible
- Clicking on the tab should load up an Piazza in an iframe. 

**Reviewers**

**Settings**
```yaml
EDXAPP_FEATURES:
  ENABLE_COMBINED_LOGIN_REGISTRATION: true

FORUM_VERSION: open-release/juniper.master
forum_version: open-release/juniper.master
```
